src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -33,6 +34,14 @@ internal static class BenchmarkRunnerClean
 
         internal static readonly IResolver DefaultResolver = new CompositeResolver(EnvironmentResolver.Instance, InfrastructureResolver.Instance);
 
+        private static IntPtr ConsoleWindowHandle;
+        private static ConsoleCancelEventHandler OnConsoleCancelEvent = delegate {
+            TaskbarManager.Instance.SetProgressState(TaskbarProgressBarState.NoProgress, ConsoleWindowHandle);
+        };
+
+        [System.Runtime.InteropServices.DllImport("kernel32.dll")]
+        private static extern IntPtr GetConsoleWindow();
```

Comment:
I would prefer to avoid adding any OS-specific sys-calls to this type (`BenchmarkRunner` is already way too big and too ugly ;) ). Instead, we could create a new `IDisposable` helper type that would do the following:

- Do nothing on non-Windows OS-es
- On Windows:
   - read console handle in the `ctor` and sign up for the Ctrl+C just once
   - unregister from the event in Dispose
   - expose two methods: `ResetProgress` and `SetProgress`

and use the helper from runner

Reply:
Done